### DISCUSSION TITLE
IBMi ports opened and services activated

### DISCUSSION THREAD

**Discussion by ghost** [codefori/vscode-ibmi#488] [Q&A] [answered]
Hi,
To use IBM softwares "Access Client Solutions" (ACS) or "Rational Developer for i" (RDi), there are some IBMi services that must be activated and corresponding ports opened.

Typically for access client solutions, I found this article : 
* TCP/IP Ports Required for IBM i Access and Related Functions : https://www.ibm.com/support/pages/tcpip-ports-required-ibm-i-access-and-related-functions

For RDi :
* Ports, protocols, and services : https://www.ibm.com/docs/en/rdfi/9.6.0?topic=overview-security-considerations#c_securityover__PortsProtocolsAndServices

But I did not find the same liste for 'Code for IBMi'. 
Does a similar 'ports and services requirements' list exists ?
I imagine the ports and services are similar with ACS and RDi ? Are they exactly the samed ? Is there more or less that must be activated ?

Thank you in advance for your light !

**Comment by worksofliam** [comment, marked as answer]
@CypBLA Hi!

Code for IBM i only uses the SSH port, which by default is 22. We don't use any other ports.

**Reply by worksofliam** [reply]
Added a note to use the SSH port.

![image](https://user-images.githubusercontent.com/3708366/151669715-d71dee5f-dcee-46d9-b3dc-7d078c0aacbe.png)

**Reply by ghost** [reply]
Thank you for you answer !

I was able use the extension launching the SSH which is not autostarting with the system.

CL command : 
```
STRTCPSVR SERVER(*SSHD)
```

Still have to investigate concerning the SQL part which still does not work but I understood I have to install 'DB2util' thanks the following resolved issue : https://github.com/halcyon-tech/vscode-ibmi/issues/513
